extensions/metacritic/src/search-metacritic.tsx

```diff
@@ -0,0 +1,258 @@
+  const image = buildImageUrl(item.images);
+  const scoreNum = normalizeScore(item.criticScoreSummary?.score);
+  const typeLabel = mediaTypeLabel(item.type);
+  const release = formatReleaseDate(item);
+  const genres = joinNames(item.genres);
+  const platforms = joinNames(item.platforms);
+  const duration = formatDuration(item.duration);
+  const badgeLine = badgesMarkdown(item);
+  const reviewsPath = item.criticScoreSummary?.url;
+  const targetUrl = reviewsPath ? `https://www.metacritic.com${reviewsPath}` : undefined;
+
+  const mdParts: string[] = [];
+  if (image) mdParts.push(`![](${image})`);
+  if (item.description) mdParts.push("\n" + item.description + "\n");
+  if (badgeLine) mdParts.push(badgeLine);
+  const markdown = mdParts.join("\n\n");
+
+  return (
+    <Detail
+      markdown={markdown}
+      metadata={<Detail.Metadata>
+        <Detail.Metadata.Label title="Title" text={item.title} />
+        {typeLabel && <Detail.Metadata.Label title="Type" text={typeLabel} />}
+        {scoreNum !== undefined && (
+          <Detail.Metadata.TagList title="Score">
+            <Detail.Metadata.TagList.Item text={`${scoreNum}`} color={scoreColor(scoreNum)} />
+          </Detail.Metadata.TagList>
+        )}
+        {release && <Detail.Metadata.Label title="Release" text={release} />}
+        {genres && <Detail.Metadata.Label title="Genres" text={genres} />}
+        {platforms && <Detail.Metadata.Label title="Platforms" text={platforms} />}
+        {item.rating && <Detail.Metadata.Label title="Rating" text={item.rating} />}
+        {item.seasonCount && item.seasonCount > 0 && (
+          <Detail.Metadata.Label title="Season Count" text={String(item.seasonCount)} />
+        )}
+        {duration && <Detail.Metadata.Label title="Duration" text={duration} />}
+      </Detail.Metadata>}
+      actions={<ActionPanel>
+        {targetUrl && <Action.OpenInBrowser title="Open on Metacritic" url={targetUrl} />}
+        <Action.CopyToClipboard title="Copy Title and Score" content={`${item.title} (${item.criticScoreSummary?.score ?? "N/A"})`} />
+      </ActionPanel>}
+    />
+  );
+}
+
+export default function Command() {
+  const [columns, setColumns] = useState(4);
+  const [searchText, setSearchText] = useState("");
+  const trimmed = searchText.trim();
+  const url = trimmed
+    ? `https://backend.metacritic.com/finder/metacritic/search/${encodeURIComponent(
+        trimmed
+      )}/web?apiKey=1MOZgmNFxvmljaQR1X9KAij9Mo4xAY3u&limit=32&offset=0`
+    : "";
+
+  const { isLoading, data } = useFetch<MCResponse>(url, {
+    execute: trimmed.length > 0,
+    keepPreviousData: true,
+    parseResponse: async (response: Response) => {
+      const json = (await response.json()) as MCResponse;
+      return json;
+    },
+  });
+
+  // Filter out people results; we only want media (games, movies, shows, etc.)
+  const items: MCItem[] = (data?.data?.items ?? []).filter((i) => i.type !== "person");
```

Comment:
2 things:

1. you are directly parsing as JSON which does not account for any possible errors,
2. you are filtering outside which IS FINE but can be paired with **1** so we get the following approach:

```suggestion
  const { isLoading, data: items } = useFetch(url, {
    execute: trimmed.length > 0,
    keepPreviousData: true,
    mapResult: (result: MCResponse) => {
      // Filter out people results; we only want media (games, movies, shows, etc.)
      const items: MCItem[] = (result.data?.items ?? []).filter((i) => i.type !== "person");
      return {
        data: items
      }
    },
    initialData: []
  });
```